docs/UpgradeGuide2.md

```diff
@@ -0,0 +1,529 @@
+  # === with 1.x ===
+  Datadog.configure do |c|
+    c.tracing.instrument :grape, error_statuses: '400,500-599'
+  end
+
+  # === with 2.0 ===
+  Datadog.configure do |c|
+    c.tracing.instrument :grape, error_status_codes: [400, 500..599]
+  end
+  ```
+
+#### GraphQL
+
+- Support changes:
+
+  - Supports `graphql-ruby` versions `>= 2.2.6`, and the below backported versions:
+    | Branch | Version |
+    | -------- | ------------------- |
+    | `2.1.x` | `>= 2.1.11, < 2.2` |
+    | `2.0.x` | `>= 2.0.28, < 2.1` |
+    | `1.13.x` | `>= 1.13.21, < 2.0` |
+  - Do **NOT** support or patch defined-based schema.
+
+- Option `schemas` becomes optional. Providing GraphQL schemas is not required. By default, every schema is instrumented.
+
+- Instrument with `GraphQL::Tracing::DataDogTrace`. Set `with_deprecated_tracer` option to `true` to rollback instrumentation with deprecated `GraphQL::Tracing::DataDogTracing`.
+
+#### GRPC
+
+- `error_handler`, `server_error_handler` and `client_error_handler` options are removed. Replace them with option `on_error`, which is invoked on both server and client side instrumentation. Merge your implementation for `server_error_handler` and `client_error_handler` to `on_error`. The implementation for `on_error` should distinguish between the server and client.
+
+  ```ruby
+  Datadog.configure do |c|
+    c.tracing.instrument :grpc, on_error: proc do |span, error|
+      if span.name == 'grpc.service'
+        # Do something for server instrumentation
+      end
+
+      if span.name == 'grpc.client'
+        # Do something for client instrumentation
+      end
+    end
+  end
+  ```
+
+#### Net/Http
+
+- `Datadog::Tracing::Contrib::HTTP::Instrumentation.after_request` has been removed.
+
+#### OpenTracing
+
+- Removed entirely. Use `ddtrace` 1.x.
+
+#### PG
+
+- Rename `error_handler` option to `on_error`. See [Error Handling](#2.0-error-handling)
+
+#### Qless
+
+- Removed entirely. Use `ddtrace` 1.x.
+
+#### Que
+
+- Rename `error_handler` option to `on_error`. See [Error Handling](#2.0-error-handling)
+
+<h4 id='2.0-instrumentation-rack'>Rack</h4>
+
+- The type for `request_queuing` option is `Boolean`. When enabled, the behaviour changes from 1 span to 2 spans. The original `http_server.queue` span rename to `http.proxy.request`, with an additional `http.proxy.queue` span representing the time spent in a load balancer queue before reaching application.
+
+  ```ruby
+  # === with 1.x ===
+  Datadog.configure do |c|
+    c.tracing.instrument :rack, request_queuing: :include_request
+    # or
+    c.tracing.instrument :rack, request_queuing: :exclude_request
+  end
+
+  # === with 2.0 ===
+  Datadog.configure do |c|
+    c.tracing.instrument :rack, request_queuing: true
+  end
+  ```
+
+  Changing the name of the top-level span (`http_server.queue` -> `http.proxy.request`) would break a lot of stuff (monitors, dashboards, notebooks). The following snippet rename the top-level span back to help migration.
```

Comment:
```suggestion
  Changing the name of the top-level span (`http_server.queue` -> `http.proxy.request`) would break functionality such as monitoring, dashboards and notebooks. The following snippet renames the top-level span back to assist with migration.
```